### PR TITLE
tests: use assertThat instead of assertEqual

### PR DESCRIPTION
**Pull request by elopio**
assertThat from testtools offers a better error message in casee of failure, and makes it easier to use a consistent order in the actual and expected arguments.

### CODE REVIEW

**Comment by elopio** [comment]
To my dear reviewers, this huge PR is for you <3
On friday I was too tired to think, and I had to find something to do in the 4 hours for the flight back. So, I made this big refactor.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -19,10 +19,13 @@ give us permission to use your contributions.
    Make a fork of Snapcraft, and create a branch named specifically for the
    feature on which you'd like to work. Make your changes there, adding new
    tests as needed, and make sure the existing tests continue to pass when your
-   changes are complete (for information about running the tests, see the
-   [HACKING][3] document).
+   changes are complete (for information see the [HACKING][3] and [TESTING][4]
+   documents).
 
-3. Squash commits into one, well-formatted commit. If you really feel like there
+3. We try to follow a consistent and readable code style. Read the
```
Comment (kyrofa) [inline CONTRIBUTING.md]: Step 2 discusses making the changes necessary. Should "make your changes while following the style guide" just be part of step 2? Or does it make that step too big?

**Comment by kyrofa** [comment]
Blech, this was mind-numbingly boring to review, I can only imagine what it was like to write. However, it is much appreciated!

@kalikiana a while back we switched to testtools, which still has an `assertEqual`. However, whereas previously the parameters were order-independent, testtools uses `assertEqual(expected, actual)` and presents failures as such. Of course, we were all over the place with our ordering, so the failure messages were often misleading. Using `assertThat(actual, Equals(expected))` is clear from the code which is actual and which is expected, and also results in a helpful failure.

File: CONTRIBUTING.md (hunk 1 above)
Comment (elopio) [inline CONTRIBUTING.md]: I prefer this in a separate step, on it's own. But I don't have a strong opinion.

**Comment by elopio** [comment]
@kalikiana I added the description.

File: CODE_STYLE.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Snapcraft code style guidelines
```
Comment (kalikiana) [inline CODE_STYLE.md]: "Coding style" is the term I would expect here

**Pull request merged**